pkg/render/node.go

```diff
@@ -56,31 +52,41 @@ type nodeComponent struct {
 }
 
 func (c *nodeComponent) Objects() ([]runtime.Object, []runtime.Object) {
-	objs := []runtime.Object{
+	objsToCreate := []runtime.Object{
 		c.nodeServiceAccount(),
 		c.nodeRole(),
 		c.nodeRoleBinding(),
 	}
+
+	var objsToDelete []runtime.Object
+
 	if c.cr.Spec.Variant == operator.TigeraSecureEnterprise {
 		// Include Service for exposing node metrics.
-		objs = append(objs, c.nodeMetricsService())
+		metricsServiceToCreate, metricsServiceToDelete := c.nodeMetricsService()
```

Comment:
I think this would be simpler to follow if we just keyed it off the existence of the `nodeMetricsPort`. In its current form, the logic for deciding if the metrics service should be created or deleted is split between here and the nodeMetricsService function. Refactoring like below will keep that logic here, and keep `nodeMetricsService`'s scope to be just "Give me a service based on my config". 

e.g., 

```
if nodeMetricsPort != nil {
  // Append returned service to objects to create
} else {
 // Append returned service to objects to delete.
}
```

Reply:
Ok